src/lib/operators/index_scan.cpp

```diff
@@ -175,8 +175,18 @@ PosList IndexScan::_scan_chunk(const ChunkID chunk_id) {
       Fail("Unsupported comparison type encountered");
   }
 
-  matches_out.reserve(matches_out.size() + std::distance(range_begin, range_end));
-  std::transform(range_begin, range_end, std::back_inserter(matches_out), to_row_id);
+  // Since index scans only work on data tables, we can gurantee that the PosList references a single chunk without
+  // checking incoming PosLists
+  matches_out.guarantee_single_chunk();
```

Comment:
Assertions made in a comment are worthless. I'd add a DebugAssert right above the call that makes sure that the input is in fact a data table. It is too easy to add support for reference tables somewhere else and forget to change this.

Reply:
I agree. The comment should not replace an assert, it's more meant like an explanation. There is an assert for the data table input in `IndexScan::_validate_input` already.

Reply:
> It is too easy to add support for reference tables somewhere else and forget to change this.

How do you guarantee that whoever adds reference support does not forget about the assumption made **here**?

A different solution would be to simply check for a data table or an input poslist with the same guarantee before you call `guarantee_single_chunk`.